scripts/validate-schemas-final.js

```diff
@@ -0,0 +1,104 @@
+   const directoryPath = './schemas';
+
+
+   const files = fs.readdirSync(directoryPath);
+
+  // Filter files based on start and end file names
+  const filteredFiles = files.filter(file => {
+    return file >= startFileName && file <= endFileName;
+  });
+
+
+    // Iterate through the filtered files
+  filteredFiles.forEach(file => {
+    // Construct the full path to the JSON schema file
+    const filePath = path.join(directoryPath, file);
+
+    try {
+      // Read and parse the JSON schema
+      const fileContent = fs.readFileSync(filePath, 'utf8');
+      const obj = JSON.parse(fileContent);
+
+      // Remove unnecessary definitions
+      delete obj.definitions['http://json-schema.org/draft-04/schema'];
+      delete obj.definitions['http://json-schema.org/draft-07/schema'];
+
+      // Compile the schema
+      const validate = ajv.validateSchema(obj);
+
+      // Check if the schema is valid
+      if (validate) {
+        console.log(`${file}: JSON Schema is valid!`);
+      } else {
+        console.error(`${file}: JSON Schema is not valid:`, ajv.errors);
+        process.exit(1);
+      }
+    } catch (error) {
+      console.error(`${file}: Error reading or parsing JSON Schema:`, error.message);
+      process.exit(1);
+    }
+  });
+}
+
+
+function validation(draft, startFileName, excludedFiles){
+
+    const Ajv = draft === 'draft04' ? require('ajv-draft-04') : require('ajv');
+    const ajv = new Ajv();
+ 
+    // Specify the path to the 'schemas' directory
+    const directoryPath = './schemas';
+ 
+ 
+    const files = fs.readdirSync(directoryPath);
+ 
+   // Filter files based on start and end file names
+   const filteredFiles = files.filter(file => {
+     return file >= startFileName && !excludedFiles.includes(file);
+   });
+ 
+ 
+    // Iterate through the filtered files
+   filteredFiles.forEach(file => {
+     // Construct the full path to the JSON schema file
+     const filePath = path.join(directoryPath, file);
+ 
+     try {
+       // Read and parse the JSON schema
+       const fileContent = fs.readFileSync(filePath, 'utf8');
+       const obj = JSON.parse(fileContent);
+ 
+       // Remove unnecessary definitions
+       delete obj.definitions['http://json-schema.org/draft-04/schema'];
+       delete obj.definitions['http://json-schema.org/draft-07/schema'];
+ 
+       // Compile the schema
+       const validate = ajv.validateSchema(obj);
+ 
+       // Check if the schema is valid
+       if (validate) {
+         console.log(`${file}: JSON Schema is valid!`);
+       } else {
+         console.error(`${file}: JSON Schema is not valid:`, ajv.errors);
+         process.exit(1);
+       }
+     } catch (error) {
+       console.error(`${file}: Error reading or parsing JSON Schema:`, error.message);
+       process.exit(1);
+     }
+   });
+}
+
+
+validationDraft04('draft04', '1.0.0-without-$id.json', '1.2.0.json');
+validation('draft07', '2.0.0-rc1.json', ['README.md', 'all.schema-store.json']);
```

Comment:
Just an idea here: would it be more future proof to have a script traversing all the files in `schema/` directory, reading every file `$shema` to determine it's version (Draft) and then validating according to that determined version? Script designed like that wouldn't have to be modified if new version of the AsyncAPI spec will land. This feature proof script must also accept list of ignored files for it to work properly.

Reply:
Thank you for this idea. Will work upon it! 🫡

Reply:
Please see the [new updated script](https://github.com/asyncapi/spec-json-schemas/pull/452/files#diff-1ee84b0b752624e8218489fead8e59e7dd93c1841a0b80418ecfa6cde1d57497), I've updated it as per your suggestion. Thank you for this! :pray: 

Marking this as resolved